oh_staff_ui/management/commands/reprocess_derivative_images.py

```diff
@@ -0,0 +1,160 @@
+                item_id=master_image.item.id,
+                file_name=thumbnail_file_name,
+                file_type=MediaFileType.objects.get(file_code="image_thumbnail"),
+                file_use="thumbnail",
+                request=request,
+            )
+            thumbnail_file.process_media_file(parent_id=master_id)
+            logger.info(f"Thumbnail image {thumbnail_file_name} created.")
+            logger.info(
+                f"Finished creating derivative images for master file {master_image.file.name}."
+            )
+
+    except (CommandError, ValueError) as ex:
+        logger.error(ex)
+        raise
+    finally:
+        # Delete the temporary files;
+        # don't throw FileNotFoundError if for some reason it doesn't exist.
+        try:
+            Path(submaster_file_name).unlink(missing_ok=True)
+            Path(thumbnail_file_name).unlink(missing_ok=True)
+        except UnboundLocalError:
+            # Swallow this, which happens when derivative creation fails
+            # so file_name variable is not defined.
+            pass
+
+
+def delete_existing_derivative_images(master_images: QuerySet) -> None:
+    """Delete existing derivative images from masters."""
+    try:
+        for master_image in master_images:
+            project_item = master_image.item
+            logger.info(
+                f"Deleting existing derivative images for project item {project_item}."
+            )
+            files_to_delete = MediaFile.objects.filter(
+                item=project_item,
+                file_type__file_code__in=["image_submaster", "image_thumbnail"],
+            )
+            for mf in files_to_delete:
+                file_name = mf.file.name
+                file_path = Path(settings.MEDIA_ROOT).joinpath(file_name)
+                mediafile_id = mf.id
+
+                # confirm that the file exists with Path
+                if file_path.exists():
+                    # delete associated file from file system
+                    mf.file.delete()
+                    # confirm that the file no longer exists
+                    if not file_path.exists():
+                        logger.info(f"Deleted {file_name} from file system.")
+                        mf.delete()
+                        logger.info(f"Deleted MediaFile {mediafile_id} from database.")
+                    else:
+                        raise CommandError(
+                            f"Failed to delete {file_name} from file system."
+                        )
+
+                else:
+                    logger.info(
+                        f"File {file_name}, for MediaFile {mediafile_id}, "
+                        "does not exist in the file system."
+                    )
+                    mf.delete()
+                    logger.info(f"Deleted MediaFile {mediafile_id} from database.")
+
+            logger.info(
+                f"Finished deleting existing derivative images for project item {project_item}."
```

Comment:
Please add the project item id to this message, for easier lookup / troubleshooting if needed - at present it just has the item title.  Something like "...for project item {project_item} ({project_item.id})" should be fine.